docs/deployment-integration.md

```diff
@@ -0,0 +1,312 @@
+It verifies signatures, timestamps public randomness, detects equivocation, and
+reports misbehavior for slashing. This is the core and only
+contract that a rollup deploys to become a BSN. The contract must be deployed
+on Babylon Genesis as it relies on direct interaction with it.
+
+## Governance Notes
+<img width="3100" height="772" alt="governance" src="https://github.com/user-attachments/assets/f1170b6c-5666-4ae9-91cc-49067f97a3a1" />
+
+
+
+Depending on the Babylon Genesis network (e.g., testnet, mainnet) you
+choose for the deployment of the rollup BSN, the network might be permissioned:
+* **Permissioned CosmWasm**: Requires governane approval for deploying a smart
+  contract. This can be granted in two ways:
+  * `MsgStoreCode` proposal: Upload a contract code. Preferred for one-time deployments
+  * `MsgUpdateParams` proposal: Add a Babylon Genesis address in an allow-list
+    for which its members can permissionlessly upload code. This is typically
+    used for contracts requiring periodic maintenance.
+  > To learn more about permissioned CosmWasm,
+  > see the [deployment
+  > guide](https://docs.babylonlabs.io/guides/governance/submit_proposals/smart_contract_deployment/) 
+  > and review past proposals in the [governance
+  > forum](https://forum.babylon.foundation/c/gov-proposals/smart-contract-proposals/15).
+* **Permissioned BSN Registration**: Requires a governance approval for
+  the registration of a BSN. This can be granted via the submission of
+  a governance proposal executing a `MsgRegisterConsumer` using the
+  metadata defined in the [Rollup BSN Registration](#rollup-bsn-registration)
+  section.
+
+  For governance guidelines, please refer to the governance forum: 
+  [Babylon Governance Guidelines](https://forum.babylon.foundation/c/gov-proposals/9/none)
+
+> **Note**: If a network uses both Permissioned CosmWasm and 
+> Permissioned BSN Registration, two separate governance proposals 
+are currently required. A unified
+> governance flow requiring only a single governance proposal is being
+> examined.
+
+For the rest of the document, we will assume a Permissionless CosmWasm and BSN
+Registration network for simplicity. Listings that would require governance
+will be highlighted appropriately.
+
+## Instantiation
+<img width="3033" height="275" alt="instantiate" src="https://github.com/user-attachments/assets/f0c6f466-a3e9-475b-a256-d79ebcfc2033" />
+
+The Rollup BSN contract is instantiated as follows:
+```shell
+INSTANT_MSG={
+  "admin": <ADMIN_ADDRESS>,
+  "bsn_id": <BSN_ID>,
+  "max_msgs_per_interval": <MAX_MSGS_PER_INTERVAL>,
+  "min_pub_rand": <MIN_PUB_RAND>,
+  "rate_limiting_interval": <RATE_LIMITING_INTERVAL>,
+  "allowed_finality_providers": [<FP_PUBKEY_1>, <FP_PUBKEY_2>] #Optional
+}
+
+babylond tx wasm instantiate $CODE_ID '$INSTANT_MSG' \
+  --from $ADMIN_KEY \
+  --chain-id $BBN_CHAIN_ID \
+  --label $LABEL \
+  --admin $ADMIN_ADDRESS \
+  --fees 100000ubbn \
+  -y
```

Comment:
We should not provide the prompt bypass flag here and allow the safety prompts to appear for users just copy pasting.